extendables/rolename.js

```diff
@@ -0,0 +1,35 @@
+const { Extendable } = require('klasa');
+const { Role } = require('discord.js');
+const roleRegex = new RegExp(/^(?:<@&)?(\\d{17,19})>?$/);
+module.exports = class extends Extendable {
+
+	constructor(...args) {
+		super(...args, ['ArgResolver'], { klasa: true });
+	}
+
+	async extend(arg, currentUsage, possible, repeat, msg) {
+		if (arg instanceof Role) return arg;
+		if (!msg.guild) return null;
+		const matches = roleRegex.exec(arg);
+		if (matches) return msg.guild.roles.get(matches[1]) || null;
```

Comment:
- You're not handling required arguments (as mentioned before) correctly.
- `return this.role(...);`